apps/judicial-system/backend/src/app/modules/case/case.service.ts

```diff
@@ -1660,8 +1647,11 @@ export class CaseService {
         updatedCourtDate.date.getTime() !== courtDate?.date.getTime()
 
       if (hasUpdatedArraignmentDate || hasUpdatedCourtDate) {
-        // New arraignment date or new court date
-        await this.addMessagesForNewCourtDateToQueue(updatedCase, user)
+        await this.eventLogService.createWithUser(
```

Comment:
This is part of the core changes. Instead of writing directly to the message queue, we trigger an event that will then handle the notifications depending on the case type.